### PR TITLE
fix #357: allowing for null instance with builders

### LINKED ISSUE #357: Add a convenience method for constructing a builder from a null value
In code paths that do some kind of create or update logic it's common to have an optional existing value.  Constructing the builder currently with a null instance results in an exception.  Can either null be allowed or a static from(instance) method be introduced?

### PR DESCRIPTION
**Pull request by shawkins**
Addresses #357 by allowing null instance values in the builder constructors.

### CODE REVIEW

File: annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java [hunk 1]
```diff
@@ -318,18 +318,25 @@ private List<Statement> getStatements(Property property, List<ClassRef> alsoImpo
...
           statements.add(new StringStatement("if (" + argumentName + " instanceof " + dunwraped + "){ this." + fieldName
               + "= new " + builderRef + "((" + dunwraped + ")" + argumentName + "); _visitables.get(\"" + fieldName
-              + "\").add(this." + fieldName + ");}"));
-
+              + "\").clear(); _visitables.get(\"" + fieldName + "\").add(this." + fieldName + ");}"));
           alsoImport.add(dunwraped);
           alsoImport.add(builderRef);
         }
+        statements.add(new StringStatement(
+            "else {  VisitableBuilder<? extends " + ((ClassRef) type).getFullyQualifiedName()
```
Comment (iocanel) [inline annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java:336]: I think that we can eliminate the `else` here as the if returns.

**Review by iocanel — APPROVED** [review]
LGTM. 
There is an optional comment that could be addressed if it makes sense.

**Comment by iocanel** [comment]
@shawkins: This will also need a rebase and conflict resolution.

File: annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java (hunk 1 above)
Comment (shawkins) [reply annotations/builder/src/main/java/io/sundr/builder/internal/functions/ToMethod.java]: Removed now.

**Comment by shawkins** [comment]
> This will also need a rebase and conflict resolution.

Should be good now.

**Pull request merged**